hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3ADataBlocks.java

```diff
@@ -849,14 +855,29 @@ long dataSize() {
       return bytesWritten;
     }
 
+    /**
+     * Does this block have unlimited space?
+     * @return true if a block with no size limit was created.
+     */
+    private boolean unlimited() {
+      return limit < 0;
+    }
+
     @Override
     boolean hasCapacity(long bytes) {
-      return dataSize() + bytes <= limit;
+      return unlimited() || dataSize() + bytes <= limit;
     }
 
+    /**
+     * {@inheritDoc}.
+     * If there is no limit to capacity, return MAX_VALUE.
+     * @return capacity in the block.
+     */
     @Override
     long remainingCapacity() {
-      return limit - bytesWritten;
+      return unlimited()
```

Comment:
remainingCapacity is long so shouldn't it be long.MAX_VALUE

Reply:
Although I see we are always casting to int. So should fine. I am assuming it is like that as we are writing the big file in disk in loop.